### PR TITLE
817 dashboard move menu

### LINKED ISSUE #817: Dashboard: move menu
Move the menu (home, images, text, time series links) from the left to the top

### PR DESCRIPTION
**Pull request by laurasootes**
This PR moves the menu from the left to the top of the page. This deviates from streamlits defaults, and requires the use of dynamic page imports. Also the ‘original menu’ has to be hidden in the settings.

These adjustments cause the changed parameters in each page to be conserved between pages. To solve this I added page-specific keys to the variable elements, which are reset when you switch pages.

Also added icons to the menu items :)

Solves #817

### CODE REVIEW

File: dianna/dashboard/Home.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import streamlit as st
 from _shared import add_sidebar_logo
 from _shared import data_directory
+from streamlit_option_menu import option_menu
+import importlib
```
Comment (SarahAlidoost) [inline dianna/dashboard/Home.py]:
```suggestion
import importlib
import streamlit as st
from _shared import add_sidebar_logo
from _shared import data_directory
from streamlit_option_menu import option_menu
```

Comment (SarahAlidoost) [reply dianna/dashboard/Home.py]: Here I just moved the `importlib` to the top to make `isort` happy. 

Can you please add `streamlit_option_menu` to `dashboard` in [setup.cfg](https://github.com/dianna-ai/dianna/blob/430be09d51c26cfc5fc7ed9b87126cb7965a544b/setup.cfg#L90)?

File: dianna/dashboard/Home.py [hunk 2]
```diff
@@ -16,26 +18,64 @@
...
+    It wraps carefully selected XAI methods in a simple, uniform interface. It's built by,
+    with and for (academic) researchers and research software engineers working on machine
+    learning projects.
 
+    ### Pages
 
-### More information
+    - <a href="/Images" target="_parent">Images</a>
+    - <a href="/Text" target="_parent">Text</a>
+    - <a href="/Time_series" target="_parent">Time series</a>
```
Comment (SarahAlidoost) [inline dianna/dashboard/Home.py:55]: Do we still need this list? I know removing this isn't a part of issue #817, but perhaps we can also remove them here.

File: dianna/dashboard/Home.py [hunk 3]
```diff
@@ -16,26 +18,64 @@
...
+    "Images": "pages.Images",
+    "Text": "pages.Text",
+    "Time series": "pages.Time_series"
+}
 
-st.image(str(data_directory / 'logo.png'))
+# Set up the top menu
+selected = option_menu(
+    menu_title=None,
+    options=["Home", "Images", "Text", "Time series"],
```
Comment (SarahAlidoost) [inline dianna/dashboard/Home.py]:
```suggestion
    options=list(pages.keys()),
```

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@laurasootes nice work :+1: I liked the small icons next to each menu option. I was able to run the dashboard, but some of the tests in [test_dashbord](https://github.com/dianna-ai/dianna/blob/main/tests/test_dashboard.py) fail on [`expect(page).to_have_title(...)`](https://github.com/dianna-ai/dianna/blob/430be09d51c26cfc5fc7ed9b87126cb7965a544b/tests/test_dashboard.py#L93).  and also some minor comments on changed files.

File: dianna/dashboard/Home.py (hunk 2 above)
Comment (laurasootes) [reply dianna/dashboard/Home.py:55]: Fair point, I think it is best to address this in [#816](https://github.com/dianna-ai/dianna/issues/816)

**Comment by laurasootes** [comment]
@SarahAlidoost thanks for your review! I implemented your code suggestions and the linting now passes. 

I looked at the error you mention for the dashboard titles. I do not know why they would arise. Also I do not get them locally and don’t see them in the logs in the github actions either so I am a bit puzzled. Could you maybe try yo run them again?

I added your comment for the pages section on the home page to #816, because I think there will be some more discussion on what should and should not be on the home page

**Comment by SarahAlidoost** [comment]
> @SarahAlidoost thanks for your review! I implemented your code suggestions and the linting now passes.
> 
> I looked at the error you mention for the dashboard titles. I do not know why they would arise. Also I do not get them locally and don’t see them in the logs in the github actions either so I am a bit puzzled. Could you maybe try yo run them again?
> 

You are right. Perhaps something related to running the tests locally on my side. I ran the dashboard test as `pytest -v -m dashboard --dashboard` on the main branch too and got the same error. Somehow the titles of pages are for example `"Text · Streamlit"` instead `"Text"` on my side :smile: . As you don't get the errors locally, I approved this PR. The failed GitHub actions are due to issue #837 .

**Review by SarahAlidoost — APPROVED** [review]

**Pull request merged**